store/storetest/group_supplier.go

```diff
@@ -1099,158 +1097,152 @@ func testPendingAutoAddChannelMembers(t *testing.T, ss store.Store) {
 	res = <-ss.Group().DeleteGroupSyncable(group.Id, channel.Id, model.GroupSyncableTypeChannel)
 	require.Nil(t, res.Err)
-	res = <-ss.Group().ChannelMembersToAdd(0)
-	require.Nil(t, res.Err)
-	require.Len(t, res.Data, 0)
+	channelMembers, err = ss.Group().ChannelMembersToAdd(0)
+	require.Nil(t, err)
+	require.Len(t, channelMembers, 0)
 
 	// reset GroupChannel and verify
 	res = <-ss.Group().UpdateGroupSyncable(&pristineSyncable)
 	require.Nil(t, res.Err)
-	res = <-ss.Group().ChannelMembersToAdd(0)
-	require.Nil(t, res.Err)
-	require.Len(t, res.Data, 1)
+	channelMembers, err = ss.Group().ChannelMembersToAdd(0)
+	require.Nil(t, err)
+	require.Len(t, channelMembers, 1)
 
 	// No result if GroupMember deleted
 	res = <-ss.Group().DeleteMember(group.Id, user.Id)
 	require.Nil(t, res.Err)
-	res = <-ss.Group().ChannelMembersToAdd(0)
-	require.Nil(t, res.Err)
-	require.Len(t, res.Data, 0)
+	channelMembers, err = ss.Group().ChannelMembersToAdd(0)
+	require.Nil(t, err)
+	require.Len(t, channelMembers, 0)
 
 	// restore group member and verify
 	res = <-ss.Group().CreateOrRestoreMember(group.Id, user.Id)
 	require.Nil(t, res.Err)
-	res = <-ss.Group().ChannelMembersToAdd(0)
-	require.Nil(t, res.Err)
-	require.Len(t, res.Data, 1)
+	channelMembers, err = ss.Group().ChannelMembersToAdd(0)
+	require.Nil(t, err)
+	require.Len(t, channelMembers, 1)
 
 	// Adding Channel (ChannelMemberHistory) should stop returning result
 	res = <-ss.ChannelMemberHistory().LogJoinEvent(user.Id, channel.Id, model.GetMillis())
 	require.Nil(t, res.Err)
-	res = <-ss.Group().ChannelMembersToAdd(0)
-	require.Nil(t, res.Err)
-	require.Len(t, res.Data, 0)
+	channelMembers, err = ss.Group().ChannelMembersToAdd(0)
+	require.Nil(t, err)
+	require.Len(t, channelMembers, 0)
 
 	// Leaving Channel (ChannelMemberHistory) should still not return result
 	res = <-ss.ChannelMemberHistory().LogLeaveEvent(user.Id, channel.Id, model.GetMillis())
 	require.Nil(t, res.Err)
-	res = <-ss.Group().ChannelMembersToAdd(0)
-	require.Nil(t, res.Err)
-	require.Len(t, res.Data, 0)
+	channelMembers, err = ss.Group().ChannelMembersToAdd(0)
+	require.Nil(t, err)
+	require.Len(t, channelMembers, 0)
 
 	// Purging ChannelMemberHistory re-returns the result
 	res = <-ss.ChannelMemberHistory().PermanentDeleteBatch(model.GetMillis()+1, 100)
 	require.Nil(t, res.Err)
-	res = <-ss.Group().ChannelMembersToAdd(0)
-	require.Nil(t, res.Err)
-	require.Len(t, res.Data, 1)
+	channelMembers, err = ss.Group().ChannelMembersToAdd(0)
+	require.Nil(t, err)
+	require.Len(t, channelMembers, 1)
 }
 
 func testTeamMemberRemovals(t *testing.T, ss store.Store) {
 	data := pendingMemberRemovalsDataSetup(t, ss)
 
 	// one result when both users are in the group (for user C)
-	res := <-ss.Group().TeamMembersToRemove()
-
-	require.Nil(t, res.Err)
-	require.Len(t, res.Data, 1)
-	teamMembers := res.Data.([]*model.TeamMember)
+	teamMembers, err := ss.Group().TeamMembersToRemove()
+	require.Len(t, teamMembers, 1)
```

Comment:
Here the `require.Nil(t, err)` is missing

Reply:
Good catch, thanks! :)

Reply:
Fixed! :)